client/v2/docker_http_.py

```diff
@@ -161,8 +164,12 @@ def __init__(
     # Ping once to establish realm, and then get a good credential
     # for use with this transport.
     self._Ping()
-    if self._authenticated:
+    if self._authentication == _BEARER:
       self._Refresh()
+    elif self._authentication == _BASIC:
+      self._creds = self._basic_creds
+    else:
+      self._creds = docker_creds.Anonymous()
```

Comment:
Might be worth a `CheckState(self._authentication == _ANONYMOUS)` for sanity-checking and documentation.